forge-gui/src/main/java/forge/player/HumanPlay.java

```diff
@@ -623,6 +624,13 @@ else if (ability.getXManaCostPaid() != null) { //ensure pre-announced X value re
             emerge = ability.getSacrificedAsEmerge();
         }
         if (!toPay.isPaid()) {
+            // if matrix still null it's effect payment
+            if (matrix == null) {
+                matrix = new ManaConversionMatrix();
+                matrix.restoreColorReplacements();
+                StaticAbilityManaConvert.manaConvert(matrix, activator, ability.getHostCard(), ability);
```

Comment:
IF its effect payment, then ability should be null (otherwise it 's problematic with ValidSA)